pkgs/development/haskell-modules/configuration-common.nix

```diff
@@ -1236,10 +1235,11 @@ self: super: {
 
   # Upstream PR: https://github.com/bgamari/monoidal-containers/pull/62
   # Bump these version bound
-  monoidal-containers = appendPatch super.monoidal-containers (pkgs.fetchpatch {
+  # For GHC 8.10 base constraint also needs relaxing.
+  monoidal-containers = doJailbreak (appendPatch super.monoidal-containers (pkgs.fetchpatch {
```

Comment:
The compiler version specific jailbreaks should go to configuration-ghc-8.10.x.nix.

Reply:
I tried but I still got the error - improved the comment to mention that.

Reply:
Strange, I'm gonna investigate that.

Reply:
When I enter `monoidal-containers = doJailbreak super.monoidal-containers;` into `configuration-ghc-8.10.x.nix`
`haskell.packages.ghc8101.monoidal-containers` builds.

Reply:
So maybe you can try that again?